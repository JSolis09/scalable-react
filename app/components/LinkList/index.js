import React from 'react';
import PropTypes from 'prop-types';

function LinkList({ links }) {
  const linkNodes = links.map(link => (
    <div key={link.id}>
      {link.url} - {link.description}
    </div>
  ));
  return <div>{linkNodes}</div>;
}

LinkList.propTypes = {
  links: PropTypes.arrayOf(
    PropTypes.shape({
      description: PropTypes.string.isRequired,
      url: PropTypes.string.isRequired,
      id: PropTypes.string.isRequired,
    }),
  ).isRequired,
};

export default LinkList;
