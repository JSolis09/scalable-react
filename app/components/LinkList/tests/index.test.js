import React from 'react';
import { shallow } from 'enzyme';
import LinkList from '../index';

describe('<LinkList />', () => {
  it('should render correctly', () => {
    const props = {
      links: [],
    };
    const linkComponent = shallow(<LinkList {...props} />);
    expect(linkComponent).toMatchSnapshot();
  });
});
