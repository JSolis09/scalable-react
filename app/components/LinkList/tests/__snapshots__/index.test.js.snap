// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LinkList /> should render correctly 1`] = `<div />`;
